### PR TITLE
Remove dependency of drake/core/** on other subdirectories.

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
This removes dependency of `drake-distro/drake/core/**` on other subdirectories. The `core` subdirectory is the first one that the `drake-distro/drake/CMakeLists.txt` visits, so must not rely on files from other, later siblings.

This was originally PR'd as #1976 and #1979.  Third time's a charm?

### CODE REVIEW

**Comment by jwnimmer-tri** [comment]
Let's see what drake00# builders say now.

**Comment by jwnimmer-tri** [comment]
@rpoyner-tri Would you code review this PR?

**Comment by jwnimmer-tri** [comment]
retest this please

**Comment by jwnimmer-tri** [comment]
Now I just need to wait for drake00# to discover that they should build this.

File: drake/core/Gradient.h [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #include <Eigen/Core>
 #include <Eigen/Dense>
 #include <unsupported/Eigen/AutoDiff>
-#include "drake/util/drakeGradientUtil.h"  // todo: pull the core tools into this file and zap the old gradient util.
+#include "drake/core/drakeGradientUtil.h"  // todo: pull the core tools into this file and zap the old gradient util.
```
Comment (rpoyner-tri) [inline drake/core/Gradient.h]: is the todo still coherent after this change?

File: drake/util/drakeGradientUtil.h [hunk 2]
```diff
@@ -1,435 +1,6 @@
...
- * \param scalar scalar to match the derivative size vector against.
- */
-template <typename Derived>
-void resizeDerivativesToMatchScalar(Eigen::MatrixBase<Derived>& mat,
-                                    const typename Derived::Scalar& scalar) {
-  internal::ResizeDerivativesToMatchScalarImpl<
-      Derived, typename Derived::Scalar>::run(mat, scalar);
-}
-}
+#include "drake/core/drakeGradientUtil.h"
```
Comment (rpoyner-tri) [inline drake/util/drakeGradientUtil.h]: Perhaps this file should also have a deprecation warning via '#pragma message' or similar.

File: drake/util/testUtil.h [hunk 3]
```diff
@@ -1,54 +1,6 @@
...
-}
-
-void valuecheck(double a, double b, double tolerance) {
-  if (std::abs(a - b) > tolerance) {
-    std::ostringstream stream;
-    stream << "Expected:\n" << a << "\nbut got:" << b << "\n";
-    throw std::runtime_error(stream.str());
-  }
-}
+#include "drake/core/testUtil.h"
```
Comment (rpoyner-tri) [inline drake/util/testUtil.h]: Perhaps this file should have a deprecation warning via '#pragma message' or similar.

File: drake/core/test/pendulum.h [hunk 4]
```diff
@@ -0,0 +1,64 @@
+#ifndef DRAKE_CORE_TEST_PENDULUM_H_
+#define DRAKE_CORE_TEST_PENDULUM_H_
+
+#include <iostream>
```
Comment (rpoyner-tri) [inline drake/core/test/pendulum.h]: unused.

File: drake/core/test/pendulum.h [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+#include <cmath>
```
Comment (rpoyner-tri) [inline drake/core/test/pendulum.h]: unused.

File: drake/core/test/pendulum.h [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+#include "drake/core/Core.h"
```
Comment (rpoyner-tri) [inline drake/core/test/pendulum.h]: None of the drake content of this include is used. Consider replacing with '#include <Eigen/Dense>'

**Comment by rpoyner-tri** [comment]
my review is done @jwnimmer-tri .

**Comment by jwnimmer-tri** [comment]
Git refuses to merge this and master successfully (it loses commits from master on the deleted files).  I will have to start over / cherry-pick into smaller PRs, I guess.

**Pull request closed without merging**